File: src/main/java/org/jukeboxmc/event/world/ChunkUnloadEvent.java
package org.jukeboxmc.event.world;

import org.jukeboxmc.event.Cancellable;
import org.jukeboxmc.world.World;
import org.jukeboxmc.world.chunk.Chunk;

/**
 * @author LucGamesYT
 * @version 1.0
 */
public class ChunkUnloadEvent extends WorldEvent implements Cancellable {

    private final Chunk chunk;
    private boolean saveChunk;

    /**
     * Creates a new {@link WorldEvent}
     *
     * @param world which represents the world which comes with this event
     */
    public ChunkUnloadEvent( World world, Chunk chunk, boolean saveChunk ) {
        super( world );
        this.chunk = chunk;
        this.saveChunk = saveChunk;
    }

    public Chunk getChunk() {
        return this.chunk;
    }

    public boolean isSaveChunk() {
        return this.saveChunk;
    }

    public void setSaveChunk( boolean saveChunk ) {
        this.saveChunk = saveChunk;
    }
}

File: JukeboxMC-API/src/main/kotlin/org/jukeboxmc/api/event/world/ChunkUnloadEvent.kt
package org.jukeboxmc.api.event.world

import org.jukeboxmc.api.event.Cancellable
import org.jukeboxmc.api.world.World
import org.jukeboxmc.api.world.chunk.Chunk

class ChunkUnloadEvent(
    world: World,
    private val chunk: Chunk,
    private var saveChunk: Boolean
) : WorldEvent(world), Cancellable {

    fun getChunk(): Chunk {
        return this.chunk
    }

    fun saveChunk(): Boolean {
        return this.saveChunk
    }

    fun setSaveChunk(saveChunk: Boolean) {
        this.saveChunk = saveChunk
    }
}